core/src/main/java/feign/java8/util/StringJoiner.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2013, Oracle and/or its affiliates. All rights reserved. DO NOT ALTER OR REMOVE
+ * COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it under the terms of the GNU
+ * General Public License version 2 only, as published by the Free Software Foundation. Oracle
+ * designates this particular file as subject to the "Classpath" exception as provided by Oracle in
+ * the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version 2 along with this work;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA or visit www.oracle.com
```

Comment:
@kdavisk6 thoughts?

Reply:
I don't think this is appropriate.  I would prefer that we implement our own version of the `join` function over copying code like this.

Regarding the problem at hand, I'm still not comfortable with creating our own "polyfills" when the JDK supports these natively.

Reply:
>  I would prefer that we implement our own version of the join
@kdavisk6 if it was just about `join` I would agree

but in the end any solution will be kind of awkward. I already detected more incompatibilities.
maybe we should conclude that `feign-10` can only be compatible with SDK >= 24.